common/authentication/postgresql/metadata.go

```diff
@@ -87,6 +102,117 @@ func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, opts Ini
 	return nil
 }
 
+// buildConnectionString builds the connection string from the metadata.
+// It supports both DSN-style and URL-style connection strings.
+// Metadata fields override existing values in the connection string.
+func (m *PostgresAuthMetadata) buildConnectionString() (string, error) {
+	if m.ConnectionString == "" {
+		return "", errors.New("connection string is required")
+	}
```

Comment:
Is an error here intended? The below method `buildDSNConnectionString` appears to build a suitable connection string

Reply:
Initially, I didn't add all the props so you couldn't build a valid connection with metadata only. Since I've now added more props to the metadata we can allow to build only from meta.